services/amundi-auth/positions_parser.py

```
@@ -0,0 +1,181 @@
+def decimal_value(value: Any) -> Decimal | None:
+    """Amundi sends money as JSON numbers; go through str to avoid float drift."""
+    if value is None or isinstance(value, bool):
+        return None
+    if isinstance(value, Decimal):
+        return value
+    if isinstance(value, int):
+        return Decimal(value)
+    if isinstance(value, float):
+        return Decimal(str(value))
+    if isinstance(value, str):
+        cleaned = value.strip().replace(" ", "").replace(" ", "").replace(",", ".")
```

Ruff (0.16.1)

[warning] String contains ambiguous ` ` (NO-BREAK SPACE). Did you mean ` ` (SPACE)?

(RUF001)